src/redisjson.rs

```diff
@@ -93,9 +129,37 @@ impl RedisJSON {
         Ok(deleted)
     }
 
-    pub fn to_string(&self, path: &str) -> Result<String, Error> {
+    pub fn to_string(&self, path: &str, format: Format) -> Result<String, Error> {
         let results = self.get_doc(path)?;
-        Ok(serde_json::to_string(&results)?)
+        let res = match format {
+            Format::JSON => serde_json::to_string(&results)?,
+            Format::BSON => return Err("Soon to come...".into()) //results.into() as Bson,
+        };
+        Ok(res)
+    }
+
+    pub fn to_json(&self, paths: &mut Vec<String>) -> Result<String, Error> {
+        let mut selector = jsonpath_lib::selector(&self.data);
+        let mut result = paths.drain(..).fold(String::from("{"), |mut acc, path| {
+            let value = match selector(&path) {
+                Ok(s) => match s.first() {
+                    Some(v) => v,
+                    None => &Value::Null,
+                },
+                Err(_) => &Value::Null,
+            };
+            acc.push('\"');
```

Comment:
It would be more readable to use `format!` instead of all these `push`es.

Reply:
what about performance?

Reply:
I guess it would be about the same (but haven’t measured it).

Reply:
I don't think so notice I'm using a single vector fold all the paths to a single JSON

Reply:
Well, the `push` can potentially do a lot of (re)allocations. This benchmark supports my gut feeling:
https://github.com/hoodie/concatenation_benchmarks-rs

Again, not important enough to argue about, whatever you decide is OK.

Reply:
Well, the push can potentially do a lot of (re)allocations. This benchmark supports my gut feeling:
https://github.com/hoodie/concatenation_benchmarks-rs

Again, not important enough to argue about, whatever you decide is OK.